addons/web/static/src/core/datepicker/datepicker.js

```diff
@@ -0,0 +1,276 @@
+
+    get formatOptions() {
+        return {
+            // Fallback to default localization format in `core/l10n/dates.js`.
+            format: this.props.format,
+            timezone: true,
+            locale: this.props.locale || this.date.locale,
+        };
+    }
+
+    get parseOptions() {
+        return {
+            // Fallback to default localization format in `core/l10n/dates.js`.
+            format: this.props.format,
+            timezone: false,
+            locale: this.props.locale || this.date.locale,
+        };
+    }
+
+    //---------------------------------------------------------------------
+    // Protected
+    //---------------------------------------------------------------------
+
+    /**
+     * Initialises formatting and parsing parameters
+     */
+    initFormat() {
+        this.defaultFormat = localization.dateFormat;
+        this.formatter = formatters.get("date");
+        this.parser = parsers.get("date");
+    }
+
+    setDate({ date, locale }) {
+        this.date = locale ? date.setLocale(locale) : date;
+    }
+
+    updateInput() {
+        try {
+            this.inputRef.el.value = this.formatter(this.date, this.formatOptions);
+        } catch (err) {
+            // Do nothing
+        }
+    }
+
+    //---------------------------------------------------------------------
+    // Bootstrap datepicker
+    //---------------------------------------------------------------------
+
+    /**
+     * Handles bootstrap datetimepicker calls.
+     * @param {string | Object} commandOrParams
+     */
+    bootstrapDateTimePicker(commandOrParams) {
+        if (typeof commandOrParams === "object") {
+            const format = luxonFormatToMomentFormat(this.props.format || this.defaultFormat);
+            const params = { ...commandOrParams, format };
+            for (const prop in params) {
+                if (params[prop] instanceof DateTime) {
+                    const luxonDate = params[prop];
+                    const momentDate = luxonDateToMomentDate(luxonDate);
+                    if (momentDate.locale() !== luxonDate.locale) {
+                        momentDate.locale(params.locale);
+                    }
+                    params[prop] = momentDate;
+                }
+            }
+            commandOrParams = params;
+        }
+        window.$(this.el).datetimepicker(commandOrParams);
+    }
+
+    //---------------------------------------------------------------------
+    // Handlers
+    //---------------------------------------------------------------------
+
+    onInputClick() {
+        this.bootstrapDateTimePicker("toggle");
+    }
+
+    onDateChange() {
+        try {
+            const date = this.parser(this.inputRef.el.value, this.parseOptions);
+            this.state.warning = date > Date.now();
+            if (!date.equals(this.props.date)) {
+                this.trigger("datetime-changed", { date });
```

Comment:
why "datetime-changed" here?

Reply:
What do you mean? You don't agree with the name of the event or it being triggered here?
- The name is completely arbitrary, I'm open to suggestions.
- It's triggered here because `onDateChange` is called on each date update (= input changed or datepicker closed), and after checking that the date value has actually changed (I don't get the point of notifying the parent with the same date).